### PR TITLE
HTTP(S) client: add support for partial Content-Type string matching, cleanup apps.c

### PR DESCRIPTION
**Pull request by DDvO**
Like with #18674, these improvements recently came up when preparing my answer at https://www.mail-archive.com/openssl-users@openssl.org/msg90965.html, where I provided a simple example HTTPS client for retrieving HTML pages.

* `OSSL_HTTP_REQ_CTX_nbio:` add support for partial `Content-Type` string matching
* code cleanup in `apps.c:` `app_http_tls_cb()` and `tls_error_hint()`

### CODE REVIEW

**Comment by t8m** [comment]
This seems to partially overlap with https://github.com/openssl/openssl/pull/18674

**Comment by DDvO** [comment]
> This seems to partially overlap with #18674

Right - it was on purpose to anticipate
```
    if (ssl_ctx == NULL) /* not using TLS */
        return bio;

```
because otherwise the guards `ssl_ctx == NULL` and `ssl_ctx != NULL` later in `app_http_tls_cb()` cannot be removed.

**Comment by DDvO** [comment]
I just tweaked the code improvements in `apps.c` 
by moving the TLS shutdown code to a helper function (which is going to be needed later).

**Comment by DDvO** [comment]
> 


> > This seems to partially overlap with #18674
> 
> Right - it was on purpose to anticipate
> 
> ```
>     if (ssl_ctx == NULL) /* not using TLS */
>         return bio;
> ```
> 
> because otherwise the guards `ssl_ctx == NULL` and `ssl_ctx != NULL` later in `app_http_tls_cb()` cannot be removed.

@t8m, is this acceptable or shall I defer all the four changes regarding `ssl_ctx == NULL` to a later PR?

**Comment by t8m** [comment]
Just wait for merge of #18674 and then rebase this please.

**Comment by DDvO** [comment]
Rebased to master after #18674 has been merged yesterday.
So the small overlap with that is resolved and reviewing can go on now.

File: doc/man3/OSSL_HTTP_REQ_CTX.pod
Comment (t8m) [inline doc/man3/OSSL_HTTP_REQ_CTX.pod:96]: Typo: should

File: crypto/http/http_client.c
Comment (t8m) [inline crypto/http/http_client.c:697]: Do not split the line here. The 80 chars limit is not hard.

File: doc/man3/OSSL_HTTP_transfer.pod
Comment (t8m) [inline doc/man3/OSSL_HTTP_transfer.pod:176]: Same typo here

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man3/OSSL_HTTP_REQ_CTX.pod
Comment (DDvO) [reply doc/man3/OSSL_HTTP_REQ_CTX.pod]: Oops, fixed this one as well as the other occurrence.

File: crypto/http/http_client.c
Comment (DDvO) [reply crypto/http/http_client.c]: Changed.

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @t8m and @paulidale

**Pull request closed without merging**
